contract-ui/tabs/settings/page.tsx

```diff
@@ -34,6 +34,13 @@ export const CustomContractSettingsTab: React.FC<
     feature: "PlatformFee",
   });
 
+  const detectedAppURI = extensionDetectedState({
+    contract,
+    feature: "AppURI",
+  });
```

Comment:
Hi Guy's, 

There is a lot of reductant code in this repo, you can reduce the number of repetitive code, If you could add ```currying``` to ```extensionDetectedState ```  and then pass ```useContract(contractAddress)``` as a single argument to pipe function with few modifications, you can use below as a pipe sample function.

```const pipe = (...func) => (value) => func.reduce((acc, fn) => [...acc, fn(value)], [])```


Then use the output of the pipe function for your if statement and for react, component part also to remove repetitive code.
It will further reduce your reductant code.